### PR TITLE
[GPU] Fix nonzero issue in constant propagate

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Fix gather_nonzero not to be marked as constant.
 - Even though count nonzero is to be turned into a constant, gather nonzero still cannot infer shape at the moment of propagate constant.

### Tickets:
 - 108008

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program.cpp [hunk 1]
```diff
@@ -631,7 +632,7 @@ void program::post_optimize_graph(bool is_internal) {
 // mark if the node is constant assuming that all dependencies are marked properly
 void program::mark_if_constant(program_node& node) {
     if (node.get_dependencies().empty() || node.is_type<prior_box>() ||
-        node.is_type<assign>() || node.is_type<read_value>()) {
+        node.is_type<assign>() || node.is_type<read_value>() || node.is_type<gather_nonzero>()) {
         return;
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/graph/program.cpp:636]: nit: What do you think to assign false explicitly as below?

```suggestion
        node.is_type<assign>() || node.is_type<read_value>() || node.is_type<gather_nonzero>()) {
        node.constant = false;
        return;
```

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/graph/program.cpp:636]: Looks good to me

**Review by e-ddykim — APPROVED** [review]
Looks good to me

File: src/plugins/intel_gpu/src/graph/program.cpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/program.cpp:636]: Because it is original behavior that only to mark "if constant" I think

**Review by andrew-k-park — APPROVED** [review]

**Pull request merged**